src/Views/BucketPanel/EntryDetail.tsx

```diff
@@ -123,6 +135,51 @@ export default function EntryDetail(props: Readonly<Props>) {
     }
   };
 
+  const handleUpload = async (values: any) => {
+    setIsUploadLoading(true);
+    try {
+      if (!uploadFile) {
+        throw new Error("Please select a file to upload");
```

Comment:
this is a very teachable case why nesting of try-catch blocks is a bad idea. You have a bug here:

![image](https://github.com/user-attachments/assets/49d21e4c-c80a-47a8-bc57-04f01239c9bb)

I've left uploadFile empty and got a confusing error.  You can do it differently:


```js

const handleUpload = async (values: any) => {
// validate data
 if (!uploadFile) {
        setUploadError("Please select a file to upload");
       return
}

setIsUploadLoading(true);
try {
// interact with the database

} catch (err) {
   setUploadError(error.message || "An unknown error occurred.");
}
setIsUploadLoading(false);

```